src/Altinn.App.Api/Extensions/WebHostBuilderExtensions.cs

```diff
@@ -15,11 +17,71 @@ public static class WebHostBuilderExtensions
     /// <param name="args">Application arguments</param>
     public static void ConfigureAppWebHost(this IWebHostBuilder builder, string[] args)
     {
+        IConfiguration? config = null;
+
         builder.ConfigureAppConfiguration(
             (_, configBuilder) =>
             {
                 configBuilder.LoadAppConfig(args);
+                config = configBuilder.Build();
             }
         );
+
+        builder.ConfigureLogging(logging =>
+        {
+            if (config is null)
+            {
+                throw new InvalidOperationException("Can't configure logging without IConfiguration");
+            }
+
+            var useOpenTelemetrySetting = config.GetValue<bool?>("AppSettings:UseOpenTelemetry");
+            if (useOpenTelemetrySetting is not true)
+            {
+                return;
+            }
+
+            logging.AddOpenTelemetry(options =>
+            {
+                options.IncludeFormattedMessage = true;
+
+                var env =
+                    Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT")
+                    ?? Environment.GetEnvironmentVariable("DOTNET_ENVIRONMENT")
+                    ?? "Production";
+
+                var (appInsightsKey, appInsightsConnectionString) = ServiceCollectionExtensions.GetAppInsightsConfig(
+                    config,
+                    env
+                );
+                if (
+                    string.IsNullOrWhiteSpace(appInsightsConnectionString) && !string.IsNullOrWhiteSpace(appInsightsKey)
+                )
+                {
+                    appInsightsConnectionString = $"InstrumentationKey={appInsightsKey}";
+                }
+
+                if (!string.IsNullOrWhiteSpace(appInsightsConnectionString))
+                {
+                    options.AddAzureMonitorLogExporter(options =>
+                    {
+                        options.ConnectionString = appInsightsConnectionString;
+                    });
+                }
+                else
+                {
+                    options.AddOtlpExporter();
+                }
+
+                builder.ConfigureServices(services =>
+                {
+                    var appBuilder = services.GetInstanceInServices<AltinnAppBuilder>();
```

Comment:
Are error messages good enough for this? If AddAltinnAppServices ends up being called after this path